### PR TITLE
Kubeadm test Isolation: #74464

### LINKED ISSUE #74464: "make test-cmd WHAT=kubeadm" performs additional tests
this PR removed the script `./hack/make-rules/test-kubeadm-cmd.sh` which can be used to run the `cmd/kubeadm/test` integration tests.

https://github.com/kubernetes/kubernetes/pull/72939

the following command is claimed to be a replacement:
```
make test-cmd WHAT=kubeadm
```

but instead it runs unrelated tests too.

expecting the above command to only show the following output (run only kubeadm tests):
```
+++ [0223 22:28:56] Building go targets for linux/amd64:
    cmd/kubeadm
found packages rand (cgo1.go) and rand2 (cgo2.go) in /home/lubo-it/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/bazel-kubernetes/external/go_sdk/doc/progs
Running tests for APIVersion: v1,admissionregistration.k8s.io/v1beta1,admission.k8s.io/v1beta1,apps/v1,apps/v1beta1,apps/v1beta2,auditregistration.k8s.io/v1alpha1,authentication.k8s.io/v1,authentication.k8s.io/v1beta1,authorization.k8s.io/v1,authorization.k8s.io/v1beta1,autoscaling/v1,autoscaling/v2beta1,autoscaling/v2beta2,batch/v1,batch/v1beta1,batch/v2alpha1,certificates.k8s.io/v1beta1,coordination.k8s.io/v1beta1,coordination.k8s.io/v1,extensions/v1beta1,events.k8s.io/v1beta1,imagepolicy.k8s.io/v1alpha1,networking.k8s.io/v1,networking.k8s.io/v1beta1,policy/v1beta1,rbac.authorization.k8s.io/v1,rbac.authorization.k8s.io/v1beta1,rbac.authorization.k8s.io/v1alpha1,scheduling.k8s.io/v1alpha1,scheduling.k8s.io/v1beta1,scheduling.k8s.io/v1,settings.k8s.io/v1alpha1,storage.k8s.io/v1beta1,storage.k8s.io/v1,storage.k8s.io/v1alpha1,
+++ [0223 22:29:16] Running tests without code coverage
ok  	k8s.io/kubernetes/cmd/kubeadm/test/cmd	29.694s
```

/assign @runyontr @spiffxp @cblecker 
@kubernetes/sig-cluster-lifecycle

### PR DESCRIPTION
**Pull request by runyontr**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Runs kubeadm without setting up API servers that are required for other tests
**Which issue(s) this PR fixes**:

Fixes #74464

**Special notes for your reviewer**:
Running against kubeadm, just tests kubeadm
```bash
make test-cmd WHAT="kubeadm"
+++ [0301 06:51:07] Running kubeadm tests
+++ [0301 06:51:12] Building go targets for darwin/amd64:
    cmd/kubeadm
Running tests for APIVersion: v1,admissionregistration.k8s.io/v1beta1,admission.k8s.io/v1beta1,apps/v1,apps/v1beta1,apps/v1beta2,auditregistration.k8s.io/v1alpha1,authentication.k8s.io/v1,authentication.k8s.io/v1beta1,authorization.k8s.io/v1,authorization.k8s.io/v1beta1,autoscaling/v1,autoscaling/v2beta1,autoscaling/v2beta2,batch/v1,batch/v1beta1,batch/v2alpha1,certificates.k8s.io/v1beta1,coordination.k8s.io/v1beta1,coordination.k8s.io/v1,extensions/v1beta1,events.k8s.io/v1beta1,imagepolicy.k8s.io/v1alpha1,networking.k8s.io/v1,networking.k8s.io/v1beta1,policy/v1beta1,rbac.authorization.k8s.io/v1,rbac.authorization.k8s.io/v1beta1,rbac.authorization.k8s.io/v1alpha1,scheduling.k8s.io/v1alpha1,scheduling.k8s.io/v1beta1,scheduling.k8s.io/v1,settings.k8s.io/v1alpha1,storage.k8s.io/v1beta1,storage.k8s.io/v1,storage.k8s.io/v1alpha1,
+++ [0301 06:51:21] Running tests without code coverage
ok  	k8s.io/kubernetes/cmd/kubeadm/test/cmd	19.722s
+++ [0301 06:51:43] TESTS PASSED
```

Running against another target skips kubeadm:
```bash
make test-cmd WHAT="authorization"
+++ [0301 06:53:03] Running kubectl tests for kube-apiserver
/Users/tom/go/src/k8s.io/kubernetes/test/cmd/../../third_party/etcd:/Users/tom/go/src/k8s.io/kubernetes/_output/local/bin/darwin/amd64:/Users/tom/google-cloud-sdk/bin:/Users/tom/.rbenv/shims:/Users/tom/go/src/github.com/kubebuilder/kudo/bin:/Users/tom/go/src/github.com/kudobuilder/kudo/cmd/kubectl-plugin:/Users/tom/.rbenv/shims/:/Users/tom/bin:/usr/local/bin:/usr/local/kubebuilder/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin
etcd --advertise-client-urls http://127.0.0.1:2379 --data-dir /var/folders/9r/yx8wxjpd4j3cnbxq_httknrc0000gn/T/tmp.sHBoIKR9 --listen-client-urls http://127.0.0.1:2379 --debug > "/dev/null" 2>/dev/null
Waiting for etcd to come up.

....

```

And running against both tests both:

```bash
make test-cmd WHAT="authorization kubeadm"
+++ [0301 06:54:24] Running kubeadm tests
+++ [0301 06:54:29] Building go targets for darwin/amd64:
    cmd/kubeadm
Running tests for APIVersion: v1,admissionregistration.k8s.io/v1beta1,admission.k8s.io/v1beta1,apps/v1,apps/v1beta1,apps/v1beta2,auditregistration.k8s.io/v1alpha1,authentication.k8s.io/v1,authentication.k8s.io/v1beta1,authorization.k8s.io/v1,authorization.k8s.io/v1beta1,autoscaling/v1,autoscaling/v2beta1,autoscaling/v2beta2,batch/v1,batch/v1beta1,batch/v2alpha1,certificates.k8s.io/v1beta1,coordination.k8s.io/v1beta1,coordination.k8s.io/v1,extensions/v1beta1,events.k8s.io/v1beta1,imagepolicy.k8s.io/v1alpha1,networking.k8s.io/v1,networking.k8s.io/v1beta1,policy/v1beta1,rbac.authorization.k8s.io/v1,rbac.authorization.k8s.io/v1beta1,rbac.authorization.k8s.io/v1alpha1,scheduling.k8s.io/v1alpha1,scheduling.k8s.io/v1beta1,scheduling.k8s.io/v1,settings.k8s.io/v1alpha1,storage.k8s.io/v1beta1,storage.k8s.io/v1,storage.k8s.io/v1alpha1,
+++ [0301 06:54:39] Running tests without code coverage
ok  	k8s.io/kubernetes/cmd/kubeadm/test/cmd	20.010s
+++ [0301 06:55:01] Running kubectl tests for kube-apiserver
/Users/tom/go/src/k8s.io/kubernetes/test/cmd/../../third_party/etcd:/Users/tom/go/src/k8s.io/kubernetes/_output/local/bin/darwin/amd64:/Users/tom/google-cloud-sdk/bin:/Users/tom/.rbenv/shims:/Users/tom/go/src/github.com/kubebuilder/kudo/bin:/Users/tom/go/src/github.com/kudobuilder/kudo/cmd/kubectl-plugin:/Users/tom/.rbenv/shims/:/Users/tom/bin:/usr/local/bin:/usr/local/kubebuilder/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin
etcd --advertise-client-urls http://127.0.0.1:2379 --data-dir /var/folders/9r/yx8wxjpd4j3cnbxq_httknrc0000gn/T/tmp.jxSOaMpe --listen-client-urls http://127.0.0.1:2379 --debug > "/dev/null" 2>/dev/null
Waiting for etcd to come up.

...
```

**Does this PR introduce a user-facing change?**:

```release-note
None
```

/sig cluster-lifecycle
/sig testing
/cc @neolit123

### CODE REVIEW

File: hack/make-rules/test-cmd.sh [hunk 1]
```diff
@@ -97,6 +97,20 @@ function create_node() {
...
+  kube::log::status "Running kubeadm tests"  
+  run_kubeadm_tests
+  # if we ONLY want to run kubeadm, then exit here.
+  if [[ $WHAT == "kubeadm" ]]; then
+    kube::log::status "TESTS PASSED"
+    exit 0
+  fi
+fi
+
+
```
Comment (neolit123) [inline hack/make-rules/test-cmd.sh]: please leave this is a single empty line.

File: hack/make-rules/test-cmd.sh [hunk 2]
```diff
@@ -106,6 +120,9 @@ create_node
 SUPPORTED_RESOURCES=("*")
 # WARNING: Do not wrap this call in a subshell to capture output, e.g. output=$(runTests)
 # Doing so will suppress errexit behavior inside runTests
+
+
+
```
Comment (neolit123) [inline hack/make-rules/test-cmd.sh]: single empty line instead of 3.

**Review by neolit123 — COMMENTED** [review]
thanks for the update @runyontr 

i seem to be getting errors when calling this:
`make test-cmd WHAT="kubeadm"`

<details>

```
+++ [0301 16:36:46] Running tests without code coverage
panic: test timed out after 2m0s

goroutine 95 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1296 +0xfd
created by time.goFunc
	/usr/local/go/src/time/sleep.go:172 +0x44

goroutine 1 [chan receive]:
testing.(*T).Run(0xc0004ac400, 0x11ca4af, 0x18, 0x123e3b0, 0x1)
	/usr/local/go/src/testing/testing.go:879 +0x383
testing.runTests.func1(0xc000572200)
	/usr/local/go/src/testing/testing.go:1119 +0x78
testing.tRunner(0xc000572200, 0xc00053de08)
	/usr/local/go/src/testing/testing.go:827 +0xbf
testing.runTests(0xc0005523a0, 0x1ce9800, 0x13, 0x13, 0x40dad3)
	/usr/local/go/src/testing/testing.go:1117 +0x2aa
testing.(*M).Run(0xc0004e0480, 0x0)
	/usr/local/go/src/testing/testing.go:1034 +0x165
main.main()
	_testmain.go:78 +0x13d

goroutine 5 [chan receive]:
k8s.io/kubernetes/vendor/k8s.io/klog.(*loggingT).flushDaemon(0x1cf6460)
	/home/lubo-it/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/klog/klog.go:941 +0x8b
created by k8s.io/kubernetes/vendor/k8s.io/klog.init.0
	/home/lubo-it/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/klog/klog.go:403 +0x69

goroutine 105 [chan receive]:
testing.(*T).Run(0xc0004ac600, 0x11c8ae0, 0x16, 0xc000189ba0, 0xc000554801)
	/usr/local/go/src/testing/testing.go:879 +0x383
k8s.io/kubernetes/cmd/kubeadm/test/cmd.TestCmdJoinDiscoveryFile(0xc0004ac400)
	/home/lubo-it/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubeadm/test/cmd/join_test.go:78 +0x187
testing.tRunner(0xc0004ac400, 0x123e3b0)
	/usr/local/go/src/testing/testing.go:827 +0xbf
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:878 +0x35c

goroutine 109 [syscall]:
syscall.Syscall6(0xf7, 0x1, 0x24ee, 0xc000089ad8, 0x1000004, 0x0, 0x0, 0x20, 0xc000189fe0, 0x0)
	/usr/local/go/src/syscall/asm_linux_amd64.s:44 +0x5
os.(*Process).blockUntilWaitable(0xc000560ea0, 0x2, 0xc000089c40, 0xc000560ea0)
	/usr/local/go/src/os/wait_waitid.go:31 +0x98
os.(*Process).wait(0xc000560ea0, 0xc0002ea480, 0x10, 0xc0002ea4d0)
	/usr/local/go/src/os/exec_unix.go:22 +0x39
os.(*Process).Wait(0xc000560ea0, 0x1240b40, 0x1240b48, 0x1240b38)
	/usr/local/go/src/os/exec.go:125 +0x2b
os/exec.(*Cmd).Wait(0xc000590840, 0x0, 0x0)
	/usr/local/go/src/os/exec/exec.go:465 +0x5b
os/exec.(*Cmd).Run(0xc000590840, 0x39, 0xc0002ea480)
	/usr/local/go/src/os/exec/exec.go:309 +0x5c
k8s.io/kubernetes/cmd/kubeadm/test/cmd.runCmdNoWrap(0x7ffee2095d0e, 0x39, 0xc0002ea480, 0x1, 0x1, 0xc0000c37d0, 0x11bee76, 0xb, 0xc000089e10, 0x40d8bf, ...)
	/home/lubo-it/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubeadm/test/cmd/util.go:37 +0xdd
k8s.io/kubernetes/cmd/kubeadm/test/cmd.RunCmd(0x7ffee2095d0e, 0x39, 0xc0002ea480, 0x1, 0x1, 0x102d2e0, 0xc0002ea420, 0xffaec0, 0xc000189dc0, 0x102d2e0, ...)
	/home/lubo-it/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubeadm/test/cmd/util.go:44 +0x77
k8s.io/kubernetes/cmd/kubeadm/test/cmd.kubeadmReset(0x7ffee2095d0e, 0x39)
	/home/lubo-it/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubeadm/test/cmd/join_test.go:24 +0x9b
k8s.io/kubernetes/cmd/kubeadm/test/cmd.TestCmdJoinDiscoveryFile.func1(0xc0004ac600)
	/home/lubo-it/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubeadm/test/cmd/join_test.go:89 +0x22a
testing.tRunner(0xc0004ac600, 0xc000189ba0)
	/usr/local/go/src/testing/testing.go:827 +0xbf
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:878 +0x35c

goroutine 113 [IO wait]:
internal/poll.runtime_pollWait(0x7ff66df43d60, 0x72, 0xc0000774e8)
	/usr/local/go/src/runtime/netpoll.go:173 +0x66
internal/poll.(*pollDesc).wait(0xc0001a6fd8, 0x72, 0xffffffffffffff01, 0x12fada0, 0x1c9d628)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:85 +0x9a
internal/poll.(*pollDesc).waitRead(0xc0001a6fd8, 0xc000558e01, 0x200, 0x200)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:90 +0x3d
internal/poll.(*FD).Read(0xc0001a6fc0, 0xc000558e00, 0x200, 0x200, 0x0, 0x0, 0x0)
	/usr/local/go/src/internal/poll/fd_unix.go:169 +0x179
os.(*File).read(0xc00000e2e0, 0xc000558e00, 0x200, 0x200, 0xc000558e00, 0x0, 0x0)
	/usr/local/go/src/os/file_unix.go:249 +0x4e
os.(*File).Read(0xc00000e2e0, 0xc000558e00, 0x200, 0x200, 0xc000077658, 0x48875c, 0xc000077660)
	/usr/local/go/src/os/file.go:108 +0x69
bytes.(*Buffer).ReadFrom(0xc000564700, 0x12fa020, 0xc00000e2e0, 0x7ff66def7178, 0xc000564700, 0x1)
	/usr/local/go/src/bytes/buffer.go:206 +0xbd
io.copyBuffer(0x12f8d80, 0xc000564700, 0x12fa020, 0xc00000e2e0, 0x0, 0x0, 0x0, 0xc0001a6cc0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:388 +0x303
io.Copy(0x12f8d80, 0xc000564700, 0x12fa020, 0xc00000e2e0, 0x406a35, 0xc0001a6e40, 0xc0000777b0)
	/usr/local/go/src/io/io.go:364 +0x5a
os/exec.(*Cmd).writerDescriptor.func1(0xc0001a6e40, 0xc0000777b0)
	/usr/local/go/src/os/exec/exec.go:279 +0x4d
os/exec.(*Cmd).Start.func1(0xc000590840, 0xc000189f60)
	/usr/local/go/src/os/exec/exec.go:400 +0x27
created by os/exec.(*Cmd).Start
	/usr/local/go/src/os/exec/exec.go:399 +0x5af

goroutine 112 [IO wait]:
internal/poll.runtime_pollWait(0x7ff66df43f00, 0x72, 0xc0004d84e8)
	/usr/local/go/src/runtime/netpoll.go:173 +0x66
internal/poll.(*pollDesc).wait(0xc0001a6f18, 0x72, 0xffffffffffffff01, 0x12fada0, 0x1c9d628)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:85 +0x9a
internal/poll.(*pollDesc).waitRead(0xc0001a6f18, 0xc00041b201, 0x565, 0x565)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:90 +0x3d
internal/poll.(*FD).Read(0xc0001a6f00, 0xc00041b29b, 0x565, 0x565, 0x0, 0x0, 0x0)
	/usr/local/go/src/internal/poll/fd_unix.go:169 +0x179
os.(*File).read(0xc00000e280, 0xc00041b29b, 0x565, 0x565, 0xc00041b200, 0xc000558c00, 0x9b)
	/usr/local/go/src/os/file_unix.go:249 +0x4e
os.(*File).Read(0xc00000e280, 0xc00041b29b, 0x565, 0x565, 0x9b, 0x0, 0x0)
	/usr/local/go/src/os/file.go:108 +0x69
bytes.(*Buffer).ReadFrom(0xc000564690, 0x12fa020, 0xc00000e280, 0x7ff66def7178, 0xc000564690, 0x1)
	/usr/local/go/src/bytes/buffer.go:206 +0xbd
io.copyBuffer(0x12f8d80, 0xc000564690, 0x12fa020, 0xc00000e280, 0x0, 0x0, 0x0, 0xc0001a6d80, 0x0, 0x0)
	/usr/local/go/src/io/io.go:388 +0x303
io.Copy(0x12f8d80, 0xc000564690, 0x12fa020, 0xc00000e280, 0x406a35, 0xc0001a6e40, 0xc0004d87b0)
	/usr/local/go/src/io/io.go:364 +0x5a
os/exec.(*Cmd).writerDescriptor.func1(0xc0001a6e40, 0xc0004d87b0)
	/usr/local/go/src/os/exec/exec.go:279 +0x4d
os/exec.(*Cmd).Start.func1(0xc000590840, 0xc000189f00)
	/usr/local/go/src/os/exec/exec.go:400 +0x27
created by os/exec.(*Cmd).Start
	/usr/local/go/src/os/exec/exec.go:399 +0x5af
FAIL	k8s.io/kubernetes/cmd/kubeadm/test/cmd	120.024s
Makefile:184: recipe for target 'test' failed

```
</details>

also the diff needs minor whitespace cleanup:
```
74807.diff:13: trailing whitespace.
  kube::log::status "Running kubeadm tests"  
74807.diff:46: trailing whitespace.
      if [[ "$pkg" != "kubeadm" ]]; then 
warning: 2 lines add whitespace errors.
```

**Comment by neolit123** [comment]
/ok-to-test
/assign @cblecker @spiffxp 
as per https://github.com/kubernetes/kubernetes/pull/72939

separating the kubeadm tests as it was before is a good change

except that these tests take a lot time to run for unknown reasons.

i've logged:
https://github.com/kubernetes/kubeadm/issues/1430

**Review by neolit123 — COMMENTED** [review]
/lgtm

File: test/cmd/legacy-script.sh [hunk 3]
```diff
@@ -397,7 +397,10 @@ runTests() {
    if [[ -n "${WHAT-}" ]]; then
     for pkg in ${WHAT}
     do 
-      record_command run_${pkg}_tests
+      # running of kubeadm is captured in hack/make-targets/test-cmd.sh
+      if [[ "$pkg" != "kubeadm" ]]; then
```
Comment (spiffxp) [inline test/cmd/legacy-script.sh]: nit
```suggestion
      if [[ "${pkg}" != "kubeadm" ]]; then 
```

**Review by neolit123 — COMMENTED** [review]
/lgtm

**Pull request merged**